enhancements/update/available-update-metadata.md

```diff
@@ -0,0 +1,171 @@
+  // field is part of spec, image is optional if version is specified and the
+  // availableUpdates field contains a matching version.
+  Image string `json:"image"`
+
+  // metadata is additional metadata associated with the release, such
+  // as associated channels (under
+  // io.openshift.upgrades.graph.release.channels) or a landing page
+  // (under url).
+  //
+  // +optional
+  Metadata map[string]string `json:"metadata,omitempty"`
+}
+```
+
+And to use that type instead of the current `Update` for
+`ClusterVersionStatus` properties.
+
+Then, in the cluster-version operator, to port existing logic like
+[available-update
+translation][cluster-version-operator-update-translation] and
+[available-update lookup][cluster-version-operator-update-lookup] to
+preserve the Cincinnati metadata.  In some cases (when an
+administrator requested an update that was not in the available update
+set), this would require an additional Cincinnati request to retrieve
+metadata about the requested release.
+
+### Metadata properties
+
+This enhancement declares the following to be well-known metadata properties:
+
+* `url` (optional), a landing page for the release.
+    For releases created by Red Hat, this will usually be the Errata for the release.
+* `io.openshift.upgrades.graph.release.channels` (optional), the comma-delimited set of channels to which the release currently belongs.
+
+Cincinnati implementation may, at their discretion, define, add, and remove additional properties as they see fit.
+
+### User Stories
+
+#### Dynamic Channel Selection
+
+Alice's cluster is on release 4.1.23, and she wants to know if she can update to 4.2.
+With interfaces like the console and `oc adm upgrade ...` exposing:
+
+```console
+$ curl -sH 'Accept:application/json' 'https://api.stage.openshift.com/api/upgrades_info/v1/graph?channel=stable-4.1&version=4.1.23' | jq -r '.nodes[] | select(.version == "4.1.23").metadata["io.openshift.upgrades.graph.release.channels"] | split(",")[]'
+prerelease-4.1
+stable-4.1
+candidate-4.2
+```
+
+she could see that, while upgrading to 4.2 was possible, it was not yet considered stable.
+And running the query again later, she could see that it had been added to `stable-4.2`.
+
+### Risks and Mitigations
+
+There is no security risk, because this is only exposing through ClusterVersion information that is already public via direct Cincinnati requests.
+
+## Design Details
+
+### Test Plan
+
+There will be no testing outside of cluster-version operator unit tests, where a test harness already exists with [mock Cincinnati responses][cluster-version-operator-available-updates-handler].
+
+### Graduation Criteria
+
+The ClusterVersion object is already GA, so there would be nothing to graduate or space to cool a preview implementation.
+
+##### Removing a deprecated feature
+
+This change would remove [the `force` property][api-force] from [`desired`][api-desired] and [`availableUpdates`][api-availableUpdates].
```

Comment:
the clusterversion spec is cleared when the update is done, and therefore we currently have the information that the update was forced.. and not we will loose any such information.. 
we are in turn loosing information here, unless the history supports storing the force, i would not be in favor of removing the force from status.desired

Reply:
> we are in turn loosing information here, unless the history supports storing the force, i would not be in favor of removing the force from status.desired

If we want to preserve forced-ness, I'm in favor of adding it to history and (if it isn't already) reported telemetry.  Telemetry could be via an alert "there is a forced update in progress".  Does this warrant a new enhancement that blocks this one?

Reply:
Closing out some details here, history already has [`Verified`][1].  That's not quite `Force`.  The interaction is that [if `Force` is set, and _verification fails_, `Verified` is false][2].  If `Force` is set, but the verifications all pass, `Verified` will be true.  So there is some data loss here, but I don't think that loss is significant (who cares if the admin was willing to waive verification failures if those failures didn't happen?).

[1]: https://github.com/openshift/api/blob/f24cbf292bdd6e6de1eec495ef88b00689d50c79/config/v1/types_cluster_version.go
[2]: https://github.com/openshift/cluster-version-operator/blob/d55d1a0a78dd05e348ec6275661b14d53e002d11/pkg/cvo/updatepayload.go#L91-L97

Reply:
I'm ok with this, given that it's not propagating verified and that will still be set.